socrata_util.py

```diff
@@ -0,0 +1,159 @@
+    def get_col_dtype_dict(self):
+        """
+        Retrieve data dictionary of a Socrata data set in the form of a dictionary,
+        with the key being the column name and the value being the column data type
+
+    	Returns:
+    		Data dictionary of a Socrata data set in the form of a dictionary,
+            with the key being the column name and the value being the column data type.
+        """
+        dataset_col_meta = self.client.get_metadata(self.dataset_id)['columns']
+        col_dtype_dict = {col['name']: col['dataTypeName'] for col in dataset_col_meta}
+        return col_dtype_dict
+
+    def mod_dtype(self, rec, col_dtype_dict=None, float_fields=None):
+        """
+        Make sure the data type of each field in the data record matches the data type
+        of the field in the Socrata data set.
+
+    	Parameters:
+    		rec: dictionary object of the data record
+            col_dtype_dict: data dictionary of a Socrata data set in the form of a dictionary,
+            with the key being the column name and the value being the column data type
+            float_fields: list of fields that should be a float
+
+    	Returns:
+    		Dictionary object of the data record, with number, string, and boolean fields
+            modified to align with the data type of the corresponding Socrata data set.
+        """
+        col_dtype_dict = col_dtype_dict or self.col_dtype_dict
+        float_fields = float_fields or self.float_fields
+
+        identity = lambda x: x
+        dtype_func = {'number': float, 'text': str, 'checkbox': bool}
+        out = {}
+        for k,v in rec.items():
+            if k in float_fields and k in col_dtype_dict:
+                out[k] = float(v)
+            elif k in col_dtype_dict:
+                if v not in [None, '']:
+                    out[k] = dtype_func.get(col_dtype_dict.get(k, 'nonexistentKey'), identity)(v)
+        out = {k:v for k,v in out.items() if k in col_dtype_dict}
+        return out
+
+    def create_new_draft(self):
+        """
+        Create a new draft of the current dataset.
+
+        Returns:
+            Draft ID of the new draft.
+        """
+        draftDataset = requests.post('https://{}/api/views/{}/publication.json'.format(self.client.domain, self.dataset_id),
+                                  auth=(self.socrata_params['username'], self.socrata_params['password']),
+                                  params={'method': 'copySchema'})
+        logger.info(draftDataset.json())
+        draftId = draftDataset.json()['id']
+        return draftId
+
+    def publish_draft(self, draftId):
+        """
+        Publish the Socrata draft specified.
+
+        Parameters:
+            draftId: 4x4 ID of the Socrata draft (e.g. x123-bc12)
+
+        Returns:
+            Response of the publish draft request.
+        """
+        time.sleep(5)
+        publishResponse = requests.post('https://{}/api/views/{}/publication.json'.format(self.client.domain, draftId),
```

Comment:
Same here. I think requests already throws exceptions if there is an error so the function might just exit anyway, if that is desired.

Reply:
Same as above.